src_py/sprite.py

```diff
@@ -1523,25 +1582,26 @@ def spritecollide(sprite, group, dokill, collided=None):
         crashed = []
         append = crashed.append
 
-        if collided:
-            for s in group.sprites():
-                if collided(sprite, s):
-                    s.kill()
-                    append(s)
-        else:
-            spritecollide = sprite.rect.colliderect
-            for s in group.sprites():
-                if spritecollide(s.rect):
-                    s.kill()
-                    append(s)
+        for group_sprite in group.sprites():
+            if collided:
+                if collided(sprite, group_sprite):
+                    group_sprite.kill()
+                    append(group_sprite)
+            else:
+                if sprite.rect.colliderect(group_sprite.rect):
+                    group_sprite.kill()
+                    append(group_sprite)
 
         return crashed
 
-    elif collided:
-        return [s for s in group if collided(sprite, s)]
-    else:
-        spritecollide = sprite.rect.colliderect
```

Comment:
optimization removed here

Reply:
Huh, I'd never heard of assigning functions to temporary variables being a python optimisation before. Looking it up though it appears to be reported several times on the internet.

I made this change because pylint starts to complain about temporary variables in a function if you have more than some limit (7?). I'll change it back and see if we are still under that limit, if not I guess we might just have to disable that warning.

Reply:
Sadly, it gets even more complicated with Cython where such an assign _defeats_ optimization (it can cause a C-callable function to degenerate into a python call).  As I understand it, Python 3.9 will hopefully make this optimization much less needed (with its optimization of the `LOAD_GLOBAL` instruction, which _tends_ to be the slowest part)

Reply:
I think Cythoniz-ing concerns are probably beyond the scope of this PR.